astropy/nddata/ccddata.py

```diff
@@ -535,10 +535,11 @@ def fits_ccddata_reader(filename, hdu=0, unit=None, hdu_uncertainty='UNCERT',
     filename : str
         Name of fits file.
 
-    hdu : int, optional
-        FITS extension from which CCDData should be initialized. If zero and
-        and no data in the primary extension, it will search for the first
-        extension with data. The header will be added to the primary header.
+    hdu : int, str, tuple of (string, int), or an HDU object, optional
```

Comment:
```suggestion
    hdu : int, str, tuple of (string, int), or BaseHDU, optional
```
Should fix the RTD warning. But actually is it really possible to pass an HDU here ?

Reply:
> But actually is it really possible to pass an HDU here ?

Looking at the code it's not possible, so we should update the type specification to:

```suggestion
    hdu : int, str, or tuple of (str, int), optional
```